.golangci.yml

```diff
@@ -98,19 +98,21 @@ linters-settings:
     # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
     max-func-lines: 30
   prealloc:
-    # XXX: we don't recommend using this linter before doing performance profiling.
-    # For most programs usage of prealloc will be a premature optimization.
-
     # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
     # True by default.
     simple: true
     range-loops: true # Report preallocation suggestions on range loops, true by default
-    for-loops: false # Report preallocation suggestions on for loops, false by default
-
+    for-loops: true # Report preallocation suggestions on for loops, false by default
+  gocritic:
+    disabled-checks:
+      - ifElseChain
+      - singleCaseSwitch
+      - appendAssign
 linters:
   enable:
     - deadcode
     - errcheck
+    - gocritic
```

Comment:
Let's not enable gocritic? It has a lot of false negatives and is going to be bothersome to maintain and provde answers to all the problems across all PRs.